File: bin/audit.py
import os
import requests

from dotenv import load_dotenv
load_dotenv('.env.local')

SUPABASE_URL = os.getenv("NEXT_PUBLIC_SUPABASE_URL")
SERVICE_ROLE = os.getenv("SUPABASE_SERVICE_ROLE_KEY")

headers = {
    "apikey": SERVICE_ROLE,
    "Authorization": f"Bearer {SERVICE_ROLE}"
}

# 1. DB pending count
res_pending = requests.get(f"{SUPABASE_URL}/rest/v1/assets?review_status=eq.pending&select=id", headers=headers)
pending_count = len(res_pending.json())

# 2. DB published count
res_published = requests.get(f"{SUPABASE_URL}/rest/v1/assets?review_status=eq.approved&select=id", headers=headers)
published_count = len(res_published.json())

# 3. DB rejected count
res_rejected = requests.get(f"{SUPABASE_URL}/rest/v1/assets?review_status=eq.rejected&select=id", headers=headers)
rejected_count = len(res_rejected.json())

# 4. Check for broken images in pending
res_pending_full = requests.get(f"{SUPABASE_URL}/rest/v1/assets?review_status=eq.pending&select=id,image_url,slug", headers=headers)
assets = res_pending_full.json()

broken_count = 0
for item in assets:
    try:
        r = requests.head(item['image_url'], timeout=5)
        if r.status_code != 200:
            broken_count += 1
    except:
        broken_count += 1

# 5. Duplicate slug check
slugs = [item['slug'] for item in assets]
duplicates = len(slugs) - len(set(slugs))

print(f"Pending Count: {pending_count}")
print(f"Published Count: {published_count}")
print(f"Rejected Count: {rejected_count}")
print(f"Broken Images in Pending: {broken_count}")
print(f"Duplicate Slugs: {duplicates}")

File: scripts/audit.js
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config({ path: '.env.local' });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

async function runAudit() {
  console.log("Starting Audit...");
  
  // 1. Check counts by review_status
  const { data: approved, error: err1 } = await supabase.from('assets').select('id', { count: 'exact', head: true }).eq('review_status', 'approved');
  const { data: pending, error: err2 } = await supabase.from('assets').select('id', { count: 'exact', head: true }).eq('review_status', 'pending');
  const { data: rejected, error: err3 } = await supabase.from('assets').select('id', { count: 'exact', head: true }).eq('review_status', 'rejected');
  
  console.log(`Approved: ${approved?.length || 0}`);
  
  // Actually count properly
  const countApproved = await supabase.from('assets').select('id', { count: 'exact', head: true }).eq('review_status', 'approved');
  const countPending = await supabase.from('assets').select('id', { count: 'exact', head: true }).eq('review_status', 'pending');
  const countRejected = await supabase.from('assets').select('id', { count: 'exact', head: true }).eq('review_status', 'rejected');
  
  console.log(`Supabase Counts -> Approved: ${countApproved.count}, Pending: ${countPending.count}, Rejected: ${countRejected.count}`);
  
  // 2. Fetch all approved to check image urls
  const { data: allApproved } = await supabase.from('assets').select('id, title, image_url, storage_key, category, tags').eq('review_status', 'approved');
  console.log(`Fetched ${allApproved.length} approved assets for integrity check.`);
  
  // Check duplicate IDs/slugs (here slug is ID usually, or title)
  const ids = new Set();
  let duplicates = 0;
  for (const a of allApproved) {
    if (ids.has(a.id)) duplicates++;
    ids.add(a.id);
  }
  console.log(`Duplicate IDs: ${duplicates}`);
  
  // Check URLs
  let brokenImages = 0;
  for (const a of allApproved) {
    let url = a.image_url;
    if (!url && a.storage_key) {
       url = `${supabaseUrl}/storage/v1/object/public/sukashi-assets/${a.storage_key}`;
    }
    if (!url) {
      brokenImages++;
    }
  }
  console.log(`Missing Image URLs/Keys: ${brokenImages}`);
  
  // 3. Categories and Tags distribution
  const catCount = {};
  const tagCount = {};
  allApproved.forEach(a => {
    catCount[a.category] = (catCount[a.category] || 0) + 1;
    (a.tags || []).forEach(t => {
      tagCount[t] = (tagCount[t] || 0) + 1;
    });
  });
  console.log("Categories:", Object.keys(catCount).length);
  console.log("Tags:", Object.keys(tagCount).length);
  
}
runAudit().catch(console.error);
